src/background/Wallet/persistence.ts

```diff
@@ -42,25 +69,130 @@ export class WalletStore extends PersistentStore<WalletStoreState> {
   }
 
   /** Prefer WalletStore['save'] unless necessary */
-  async encryptAndSave(
+  private async encryptAndSave(
     id: string,
-    encryptionKey: string,
+    credentials: Credentials,
     record: WalletRecord
   ) {
-    const encryptedRecord = await Model.encryptRecord(encryptionKey, record);
-    this.setState((state) =>
+    const encryptedRecord = await Model.encryptRecord(
+      credentials.encryptionKey,
+      record
+    );
+    await this.setState((state) =>
       produce(state, (draft) => {
         draft[id] = encryptedRecord;
       })
     );
     this.lastRecord = record;
   }
 
-  async save(id: string, encryptionKey: string, record: WalletRecord) {
+  async save(id: string, credentials: Credentials, record: WalletRecord) {
     if (this.lastRecord === record) {
       return;
     }
-    await this.encryptAndSave(id, encryptionKey, record);
+    await this.encryptAndSave(id, credentials, record);
+  }
+
+  async createBackup(id: string) {
```

Comment:
what do you think about calling this param `userId` to make to purpose more clear?